src/meltano/core/project_plugins_service.py

```diff
@@ -400,48 +429,122 @@ def update_environment_plugin(self, plugin: EnvironmentPluginConfig) -> None:
 
             return p_outdated
 
-    def get_parent(self, plugin: ProjectPlugin) -> Optional[ProjectPlugin]:
-        """Get plugin's parent plugin.
+    def _get_parent_from_discovery(self, plugin: ProjectPlugin) -> ProjectPlugin:
+        """Get the parent plugin from discovery.yml.
 
         Args:
             plugin: The plugin to get the parent of.
 
         Returns:
-            The parent plugin or None if the plugin has no parent.
+            The parent plugin.
+
+        Raises:
+            PluginParentNotFoundError: If the parent plugin is not found.
+        """
+        try:
+            return self.discovery_service.get_base_plugin(plugin)
+        except (PluginNotFoundError, VariantNotFoundError) as err:
+            if plugin.inherit_from:
+                raise PluginParentNotFoundError(plugin, err) from err
+
+            raise
+
+    def _get_parent_from_hub(self, plugin: ProjectPlugin) -> ProjectPlugin:
+        """Get the parent plugin from the hub.
+
+        Args:
+            plugin: The plugin to get the parent of.
+
+        Returns:
+            The parent plugin.
+
+        Raises:
+            PluginParentNotFoundError: If the parent plugin is not found.
+        """
+        try:
+            return self.hub_service.get_base_plugin(plugin, variant_name=plugin.variant)
+        except PluginNotFoundError as err:
+            if plugin.inherit_from:
+                raise PluginParentNotFoundError(plugin, err) from err
+
+            raise
+
+    def find_parent(
```

Comment:
This is where the magic of finding a plugin definition from available sources happens

Reply:
I created https://github.com/meltano/meltano/issues/5970